index.js

```diff
@@ -27,11 +29,16 @@ function levelPlugin (fastify, opts, next) {
   }
   opts.options = opts.options || {}
 
-  fastify
-    .decorate('level', levelMore(opts.name, opts.options))
-    .addHook('onClose', close)
+  // asynchronous mode to catch possible init errors
+  levelMore(opts.name, opts.options)
+    .then((db) => {
+      fastify
+        .decorate('level', db)
+        .addHook('onClose', close)
 
-  next()
+      next()
+    })
+    .catch(next)
```

Comment:
Dropping Node 6 is not needed here.

```js
levelMore(opts.name, opts.options, function (err, db) {
  if (err) {
    next(err)
    return 
  }
  fastify.decorate('level', db)
  fastify.addHook('onClose', close)
  next()
})
```

Reply:
But [leveldown itself requires >= 8.6.0](https://github.com/Level/leveldown/pull/552).

I can revert the change, but I'll also have to remove `util.promisify()`.

Reply:
Oh ok. I would prefer to avoid promisify anyway as there is no need to use promises here.